include/dlpack/dlpack.h

```diff
@@ -200,6 +209,15 @@ typedef struct {
   int64_t* strides;
   /*! \brief The offset in bytes to the beginning pointer to data */
   uint64_t byte_offset;
+  /*! \brief Mark the data readonly. */
+  uint8_t readonly;
+  /*!
+  * \brief Endianness of the data. 1 for non-native endianness and
+  *  0 for native endianness.
+  */
+  uint8_t endianness;
+  /*! \brief Alignment of the data. */
+  uint32_t alignment;
```

Comment:
These changes still need to be discussed. Since we are breaking the ABI by adding the version info, it would be good to discuss these here and add them too (so we don't need to break the ABI again)